src/transaction.rs

```diff
@@ -0,0 +1,254 @@
+                key.len() as size_t,
+            ));
+            Ok(())
+        }
+    }
+
+    /// Read a key inside a transaction
+    ///
+    /// ColumnFamilyHandle: default
+    pub fn get_opt(
+        &self,
+        key: &[u8],
+        mut readopts: ReadOptions,
+    ) -> Result<Option<DBVector>, Error> {
+        if readopts.inner.is_null() {
+            return Err(Error::new(
+                "Unable to create RocksDB read options. \
+                 This is a fairly trivial call, and its \
+                 failure may be indicative of a \
+                 mis-compiled or mis-loaded RocksDB \
+                 library."
+                    .to_owned(),
+            ));
+        }
+
+        if let Some(snapshot) = self.snapshot {
+            readopts.set_snapshot(snapshot)
+        };
+
+        unsafe {
+            let mut val_len: size_t = 0;
+            let val = ffi_try!(ffi::rocksdb_transaction_get(
+                self.inner,
+                readopts.inner,
+                key.as_ptr() as *const c_char,
+                key.len() as size_t,
+                &mut val_len,
+            )) as *mut u8;
+            if val.is_null() {
+                Ok(None)
+            } else {
+                Ok(Some(DBVector::from_c(val, val_len)))
+            }
+        }
+    }
+
+    /// Read a key inside a transaction
+    ///
+    /// ReadOptions: default
+    /// ColumnFamilyHandle: default
+    pub fn get(&self, key: &[u8]) -> Result<Option<DBVector>, Error> {
+        self.get_opt(key, ReadOptions::default())
+    }
+
+    /// Read a key inside a transaction
+    pub fn get_cf_opt(
+        &self,
+        cf: ColumnFamily,
+        key: &[u8],
+        mut readopts: ReadOptions,
+    ) -> Result<Option<DBVector>, Error> {
+        if readopts.inner.is_null() {
+            return Err(Error::new(
+                "Unable to create RocksDB read options. \
+                 This is a fairly trivial call, and its \
+                 failure may be indicative of a \
+                 mis-compiled or mis-loaded RocksDB \
+                 library."
+                    .to_owned(),
+            ));
+        }
+
+        if let Some(snapshot) = self.snapshot {
+            readopts.set_snapshot(snapshot)
+        };
+
+        unsafe {
+            let mut val_len: size_t = 0;
+            let val = ffi_try!(ffi::rocksdb_transaction_get_cf(
+                self.inner,
+                readopts.inner,
+                cf.inner,
+                key.as_ptr() as *const c_char,
+                key.len() as size_t,
+                &mut val_len,
+            )) as *mut u8;
+            if val.is_null() {
+                Ok(None)
+            } else {
+                Ok(Some(DBVector::from_c(val, val_len)))
+            }
+        }
+    }
+
+    /// Read a key inside a transaction
+    ///
+    /// ReadOptions: default
+    pub fn get_cf(&self, cf: ColumnFamily, key: &[u8]) -> Result<Option<DBVector>, Error> {
```

Comment:
Can you make `key` generic? eg: `K: AsRef<[u8]>` See #253 for some examples of similar changes. Also, can `cf` be a reference, eg: `cf: &ColumnFamily`?